contracts/Graph.sol

```diff
@@ -1,103 +1,177 @@
-pragma solidity 0.8;
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.18;
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
```

Comment:
Safe math is no longer needed in 0.8.x https://stackoverflow.com/questions/70074736/do-you-need-safemath-in-solidity-version-0-8-and-if-not-can-you-still-import